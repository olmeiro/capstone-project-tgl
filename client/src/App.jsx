import React from 'react'
import { AppRouter } from './routes/AppRouter'
import { HashRouter } from 'react-router-dom'

export default function App () {
  return (
     <HashRouter>
       <AppRouter />
     </HashRouter>
  )
}
