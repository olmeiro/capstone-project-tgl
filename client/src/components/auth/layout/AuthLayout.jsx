import React from 'react'
import PropTypes from 'prop-types'

import imagePath from '../../../../assets/logo_Team_International.png'

export const AuthLayout = ({ children, title = '' }) => {
  return (
    <section className="w-full  flex flex-row sm:flex-col min-h-screen">
      <header className="w-full min-h-full flex flex-row sm:flex-col md:flex-row justify-between p-6 items-center h-28 bg text-team-green">
        <div className="mt-2">
          <img className="w-24" src={imagePath} alt='logo team' />
        </div>
      </header>
      <main className="flex flex-row min-h-[74.2vh] sm:flex-col md:flex-row">

        <section className="w-1/2 sm:w-full flex flex-col justify-center p-10 place-items-center">
          <h1 className='text-7xl sm:text-xl p-6'>Red Social TGL</h1>
          <img className="bg w-56 rounded-lg p-9 mt-9 mb-9" src={imagePath} alt='logo team' />
          <footer>
            <p className='text-sm text-clip whitespace-normal'>Red social Top Gun Lab. Realiza tus proyectos en compa√±ia de los mejores.</p>
          </footer>
        </section>

        <section className="w-1/2 sm:w-full flex flex-col justify-center border-b-team-green">
          <h4 className="text-center text-5xl p-3 text-team-blue  ">{title}</h4>
          {children}
        </section>
      </main>
      <footer className="max-w-full mt-[10] bg-team-blue h-20 text-lg font-medium text-slate-200 flex text-center justify-center items-center">
        <p>&copy;2022 tgl_group. All Rights Reserved.</p>
      </footer>
    </section>
  )
}

AuthLayout.propTypes = {
  children: PropTypes.node,
  title: PropTypes.string
}
