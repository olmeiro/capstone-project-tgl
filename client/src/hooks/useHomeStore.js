import React from 'react'
import { useDispatch, useSelector } from 'react-redux'
import { socialApi } from '../api'
import { checkFavorites, getFavorites, actionKeepLastUserVisitesOnRedux, getPostsToHome, getFriendsFromFriends, checkComments, getUserSearched, checkEmptySearchBar, setPathReference, setLastUserVisited, setInfoFromUserLoggedIn, setcheckInfoUser, setPostsFromUserLoggedIn } from '../store'

export const useHomeStore = () => {
  const dispatch = useDispatch()
  const { user } = useSelector(state => state.auth)
  const userId = user.id

  const entire = socialApi.get(`/user/byid/${userId}`).then(response => response.data.body)
  let friends = entire.then(user => user.friends)

  const getPostsToHomeHook = async () => {
    friends = await friends
    let postsTotalFromFriends = friends.map(async friendId => await socialApi.get(`/posts/byuser/${friendId}`))
    postsTotalFromFriends = Promise.all(postsTotalFromFriends)
    postsTotalFromFriends = await postsTotalFromFriends
    postsTotalFromFriends = postsTotalFromFriends.map(posts => posts.data.body).flat()
    dispatch(getPostsToHome(postsTotalFromFriends))
  }
  const getFriendsFromFriendsHook = async () => {
    const wholeUser = await socialApi.get(`/user/byid/${userId}`)
    const friendsId = wholeUser.data.body.friends
    let suggestion = friendsId.map(async id => {
      const response = await socialApi.get(`/user/byid/${id}`)
      const data = response.data
      const suggestion = data.body.friends
      return suggestion
    })
    suggestion = Promise.all(suggestion)
    suggestion = await suggestion
    suggestion = suggestion.flat()
    suggestion = suggestion.filter((friendid, index) => suggestion.indexOf(friendid) == index)
    suggestion = suggestion.map(async id => {
      const response = await socialApi.get(`/user/byid/${id}`)
      const data = response.data
      const suggestion = data.body
      return suggestion
    })
    suggestion = Promise.all(suggestion)
    suggestion = await suggestion
    dispatch(getFriendsFromFriends(suggestion))
  }
  const addFriendshipHook = async (friendId) => {
    await socialApi.post('/friends/', { userId, friendId })
  }
  const likeAPost = async (id) => {
    await socialApi.put('/posts/', { id, likes: 3 })
  }
  const makeAComment = async (postId, comment) => {
    await socialApi.post('/comments/', { comment, postId, userId })
  }
  const getCommentsByPost = async (postid) => {
    let comments = await socialApi.get(`/comments/bypost/${postid}`)
    comments = comments.data.body
    comments = comments.map(async comment => {
      const userId = comment.UserId
      let user = await socialApi.get(`/user/byid/${userId}`)
      user = user.data.body
      comment.user = user
      return comment
    })
    comments = await Promise.all(comments)
    return comments
  }
  const deleteComment = async (commentId) => {
    await socialApi.delete(`/comments/${commentId}`)
  }
  const checkCommentsHook = () => {
    dispatch(checkComments())
  }
  const searchUserByAlias = async (aliasFromBody) => {
    let foundUsers = null
    if (aliasFromBody.trim() != '') {
      const response = await socialApi.get('/user/all')
      const users = response.data.body
      foundUsers = users.filter(user => user.alias.toLowerCase().includes(aliasFromBody && aliasFromBody.toLowerCase()))
    }
    dispatch(getUserSearched(foundUsers))
  }
  const checkEmptySearchBarHook = () => {
    dispatch(checkEmptySearchBar())
  }
  const sendPathHook = (pathReference) => {
    dispatch(setPathReference(pathReference))
  }
  const setLastUserVisitedHook = async (userAlias, userId) => {
    const responseUser = await socialApi.get(`user/byalias/${userAlias}`)
    const user = responseUser.data.body
    const responsePosts = await socialApi.get(`/posts/byuser/${userId}`)
    const posts = responsePosts.data.body
    user.posts = posts
    localStorage.setItem('lastUserVisited', JSON.stringify(user))
    const lastUserVisited = JSON.parse(localStorage.getItem('lastUserVisited'))
    dispatch(setLastUserVisited(lastUserVisited))
  }
  const keepLastUserVisitesOnRedux = async () => {
    const lastUserVisited = JSON.parse(localStorage.getItem('lastUserVisited'))
    dispatch(actionKeepLastUserVisitesOnRedux(lastUserVisited))
  }
  const getInfoFromTheUserLoggedIn = async () => {
    const user = await entire
    const responsePosts = await socialApi.get(`/posts/byuser/${userId}`)
    const posts = responsePosts.data.body
    const newUser = {
      ...user,
      posts
    }
    dispatch(setInfoFromUserLoggedIn(newUser))
  }
  const checkInfoUserHook = async () => {
    dispatch(setcheckInfoUser())
  }
  const getPostsFromUserLoggedIn = async () => {
    const responsePosts = await socialApi.get(`/posts/byuser/${userId}`)
    const posts = responsePosts.data.body
    dispatch(setPostsFromUserLoggedIn(posts))
  }
  const addToFavorites = async (postFavoriteId, userLogeadoId) => {
    await socialApi.post(`/favorites?postId=${postFavoriteId}&userId=${userLogeadoId}`)
  }
  const getFavoritesHook = async () => {
    const response = await socialApi.get(`/favorites/${userId}`)
    const favorites = response.data.body
    dispatch(getFavorites(favorites))
  }
  const deleteFavoriteHook = async (postId) => {
    await socialApi.delete(`/favorites?postId=${postId}&userId=${userId}`)
  }
  const checkFavoritesHook = async () => {
    dispatch(checkFavorites())
  }

  return {
    getPostsToHomeHook,
    getFriendsFromFriendsHook,
    addFriendshipHook,
    likeAPost,
    makeAComment,
    getCommentsByPost,
    deleteComment,
    checkCommentsHook,
    searchUserByAlias,
    checkEmptySearchBarHook,
    sendPathHook,
    setLastUserVisitedHook,
    getInfoFromTheUserLoggedIn,
    checkInfoUserHook,
    getPostsFromUserLoggedIn,
    keepLastUserVisitesOnRedux,
    addToFavorites,
    getFavoritesHook,
    deleteFavoriteHook,
    checkFavoritesHook
  }
}
