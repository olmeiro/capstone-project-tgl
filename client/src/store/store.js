import { configureStore } from '@reduxjs/toolkit'

import { authSlice, profileSlice, homeSlice, friendSlice } from './index'

export const store = configureStore({
  reducer: {
    auth: authSlice.reducer,
    profile: profileSlice.reducer,
    home: homeSlice.reducer,
    friends: friendSlice.reducer
  },
  middleware: (getDefaultMiddleware) => getDefaultMiddleware({
    serializableCheck: false
  })
})
