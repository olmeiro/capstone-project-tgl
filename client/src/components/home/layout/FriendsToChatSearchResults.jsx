import photoDefault from "../../../../assets/photoDefault.png"
import { socialApi } from "../../../api"
import { useHomeStore } from "../../../hooks/useHomeStore"

const FriendsToChatSearchResults = ({ foundFriends, userId, handleClearInput }) => {

    const { setCurrentChatHook, chekChangeChatHook } = useHomeStore()

    const handleAddConversation = async (receiverId) => {
        const members = {
            senderId: userId,
            receiverId
        }
        await socialApi.post("/conversations", members)
        const response = await socialApi.get(`/conversations/${userId}`)
        const conversationsUserId = response.data.body;
        const currentChat = conversationsUserId.filter(conversation=> conversation.members.includes(receiverId))
        setCurrentChatHook(currentChat[0])
        handleClearInput()
        chekChangeChatHook()
    }


    return (
        <div>
            {
                foundFriends.length != 0
                    ? foundFriends.map(friend => {
                        return (
                            <div
                                onClick={() => handleAddConversation(friend.id)}
                                className='cursor-pointer relative rounded-lg px-2 py-2 flex items-center space-x-3 hover:border-gray-400 focus-within:ring-2 mb-3 hover:bg-gray-200'
                            >
                                <div className='flex-shrink-0 '>
                                    <img
                                        src={friend && friend.photoProfile ? friend.photoProfile : photoDefault}
                                        className='h-10 w-10 rounded-full'
                                    />
                                </div>

                                <div className='flex-1 min-w-0'>
                                    <div className='flex items-center justify-between'>
                                        <p className='text-sm font-bold text-team-dark'>
                                            {friend && friend.alias}
                                        </p>
                                    </div>
                                </div>
                            </div>
                        )
                    })
                    : <div>amigo no existe</div>
            }
        </div>

    )

}

export default FriendsToChatSearchResults