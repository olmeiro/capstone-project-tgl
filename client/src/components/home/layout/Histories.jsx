import React, { useEffect } from 'react'

import { useHomeStore } from '../../../hooks/useHomeStore'
import { Card } from './card/Card'
import { useSelector } from 'react-redux'

export const Histories = () => {

  const { getPostsToHomeHook, likeAPost, getInfoFromTheUserLoggedIn, getFavoritesHook } = useHomeStore()
  const { posts } = useSelector(state => state.home)

  useEffect(() => {
    getPostsToHomeHook()
    getInfoFromTheUserLoggedIn()
    getFavoritesHook()
  }, [])

  return (
    <div className="flex flex-col basis-1 md:basis-1/3 scroll-smooth">
      {posts.length === 0
        ? null
        : posts.map((post) => {
          const date = new Date(post.date)
          const options = {
            timeStyle: 'short',
            dateStyle: 'full'
          }
          const colFormatter = Intl.DateTimeFormat('es-CO', options).format(date)
          return (<Card
              userId={post.UserId}
              postId={post.id}
              likeAPost={likeAPost}
              photo={post.photo}
              description={post.description}
              likes={post.likes}
              date={colFormatter}
              key={post.id}
            />)
        })}
    </div>
  )
}
