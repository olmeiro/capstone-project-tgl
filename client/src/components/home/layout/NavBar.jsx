import React, { Fragment, useState, useEffect } from 'react'
import { useSelector } from 'react-redux'
import { Disclosure, Menu, Transition } from '@headlessui/react'
import { UserIcon, MenuIcon, XIcon } from '@heroicons/react/outline'
import { Tooltip } from 'flowbite-react'

import { useAuthStore } from '../../../hooks'
import { useHomeStore } from '../../../hooks/useHomeStore'
import imagePath from '../../../../assets/logo_Team_International.png'
import { Link } from 'react-router-dom'

const userMock = {
  name: 'Tom Cook',
  email: 'tom@example.com',
  imageUrl:
    'https://images.unsplash.com/photo-1472099645785-5658abf4ff4e?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=facearea&facepad=2&w=256&h=256&q=80'
}
const navigation = [
  { name: 'Team', href: '/', current: false },
  { name: 'Perfil', href: '/profile', current: false },
  { name: 'Amigos', href: '/friends', current: false },
  { name: 'Favoritos', href: '/favorites', current: false },
  { name: 'Mensajes', href: '/messenger', current: false }
]
const userNavigation = [
  { name: 'Salir', href: '#' }
]

function classNames(...classes) {
  return classes.filter(Boolean).join(' ')
}

export function NavBar() {
  const { startLogout } = useAuthStore()
  const [search, setSearch] = useState('')

  const { searchUserByAlias, checkEmptySearchBarHook } = useHomeStore()
  const { pathReference } = useSelector((state) => state.home)
  const { alias, photoProfile, email } = useSelector((state) => state.profile)

  const handleSearch = (e) => {
    setSearch(e.target.value)
  }
  const handleClearSearchBar = () => {
    setSearch('')
  }
  useEffect(() => {
    (async () => {
      await searchUserByAlias(search)
      checkEmptySearchBarHook()
    })()
  }, [search])

  return (
    <>
      <Disclosure as="nav" className="bg-team-dark">
        {({ open }) => (
          <>
            <div className="max-w-7xl mx-auto px-4 sm:px-6 lg:px-8">
              <div className="flex items-center justify-between h-24">
                <div className="flex items-center">
                  <div className="flex-shrink-0">
                    <img
                      className="h-8 sm:w-[2.4rem] md:w-24 rounded-none bg-team-dark"
                      src={imagePath}
                      alt="Workflow"
                    />
                  </div>

                  <div className="hidden md:block">
                    <div className="ml-10 flex items-baseline space-x-4">
                      {navigation.map((item) => {
                        return (
                          pathReference == item.href
                            ? <Link
                              to={item.href}
                              key={item.name}
                              className={classNames(
                                item.current
                                  ? 'bg-gray-900 text-white'
                                  : 'text-team-brown hover:bg-gray-700 hover:text-white',
                                'px-3 py-2 rounded-md text-sm font-medium'
                              )}
                              aria-current={item.current ? 'page' : undefined}
                            >
                              {item.name}
                            </Link>
                            : <Link
                              to={item.href}
                              key={item.name}
                              className={classNames(
                                item.current
                                  ? 'bg-gray-900 text-white'
                                  : 'text-team-green hover:bg-gray-700 hover:text-white',
                                'px-3 py-2 rounded-md text-sm font-medium'
                              )}
                              aria-current={item.current ? 'page' : undefined}
                            >
                              {item.name}
                            </Link>
                        )
                      })}
                    </div>
                  </div>
                </div>

                {/* search bar */}
                {pathReference === '/'
                  ? (
                  <div>
                    <div className="p-3">
                      <div className="relative">
                        <div className="flex absolute inset-y-0 left-0 items-center pl-3 pointer-events-none">
                          <svg
                            className="w-5 h-5 text-gray-500 dark:text-gray-400"
                            aria-hidden="true"
                            fill="currentColor"
                            viewBox="0 0 20 20"
                            xmlns="http://www.w3.org/2000/svg"
                          >
                            <path
                              fillRule="evenodd"
                              d="M8 4a4 4 0 100 8 4 4 0 000-8zM2 8a6 6 0 1110.89 3.476l4.817 4.817a1 1 0 01-1.414 1.414l-4.816-4.816A6 6 0 012 8z"
                              clipRule="evenodd"
                            ></path>
                          </svg>
                        </div>
                        <input
                          value={search}
                          onChange={(e) => handleSearch(e)}
                          type="text"
                          id="input-group-search"
                          className="block p-2 pl-10 w-full text-sm text-gray-900 bg-gray-50 rounded-lg border border-gray-300 focus:ring-blue-500 focus:border-blue-500 dark:bg-gray-600 dark:border-gray-500 dark:placeholder-gray-400 dark:text-white dark:focus:ring-blue-500 dark:focus:border-blue-500 mr-14"
                          placeholder="Buscar usuario"
                        />
                        <div className="flex absolute inset-y-0 right-0 items-center pl-3">
                          <button onClick={() => handleClearSearchBar()}>
                            <svg
                              xmlns="http://www.w3.org/2000/svg"
                              className="w-5 h-5 text-gray-500 dark:text-gray-400 mr-2"
                              fill="none"
                              viewBox="0 0 24 24"
                              stroke="currentColor"
                              strokeWidth="2"
                            >
                              <path
                                strokeLinecap="round"
                                strokeLinejoin="round"
                                d="M10 14l2-2m0 0l2-2m-2 2l-2-2m2 2l2 2m7-2a9 9 0 11-18 0 9 9 0 0118 0z"
                              />
                            </svg>
                          </button>
                        </div>
                      </div>
                    </div>
                  </div>
                    )
                  : null}

                <div className="hidden md:block">
                  <div className="ml-4 flex items-center md:ml-6">
                    <Tooltip
                      content={email}
                    >
                      <div
                        className="flex bg-gray-800 p-1 rounded-full text-team-green hover:text-white focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-offset-gray-800 focus:ring-white"
                      >
                        <span className="sr-only">View notifications</span>
                        <p className="ml-3">Bienvenido {alias}</p>
                        <UserIcon
                          className="h-6 w-6 ml-1 mr-3 rounded-md"
                          aria-hidden="true"
                        />
                      </div>
                    </Tooltip>

                    {/* Profile dropdown */}
                    <Menu as="div" className="ml-3 relative">
                      <div>
                        <Menu.Button className="max-w-xs bg-gray-800 rounded-full flex items-center text-sm focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-offset-gray-800 focus:ring-white">
                          <span className="sr-only">Open user menu</span>
                          <img
                            className="h-8 w-8 rounded-full"
                            src={
                              photoProfile !== ''
                                ? photoProfile
                                : userMock.imageUrl
                            }
                            alt=""
                          />
                        </Menu.Button>
                      </div>
                      <Transition
                        as={Fragment}
                        enter="transition ease-out duration-100"
                        enterFrom="transform opacity-0 scale-95"
                        enterTo="transform opacity-100 scale-100"
                        leave="transition ease-in duration-75"
                        leaveFrom="transform opacity-100 scale-100"
                        leaveTo="transform opacity-0 scale-95"
                      >
                        <Menu.Items className="origin-top-right absolute right-0 mt-2 w-48 rounded-md shadow-lg py-1 bg-white ring-1 ring-black ring-opacity-5 focus:outline-none">
                          {userNavigation.map((item) => (
                            <Menu.Item key={item.name}>
                              {({ active }) => (
                                <a
                                  href={item.href}
                                  className={classNames(
                                    active ? 'bg-gray-100' : '',
                                    'block px-4 py-2 text-sm text-gray-700'
                                  )}
                                  onClick={startLogout}
                                >
                                  {item.name}
                                </a>
                              )}
                            </Menu.Item>
                          ))}
                        </Menu.Items>
                      </Transition>
                    </Menu>
                  </div>
                </div>
                <div className="-mr-2 flex md:hidden">
                  {/* Mobile menu button */}
                  <Disclosure.Button className="bg-gray-800 inline-flex items-center justify-center p-2 rounded-md text-gray-400 hover:text-white hover:bg-gray-700 focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-offset-gray-800 focus:ring-white">
                    <span className="sr-only">Open main menu</span>
                    {open
                      ? (
                        <XIcon className="block h-6 w-6" aria-hidden="true" />
                      )
                      : (
                        <MenuIcon className="block h-6 w-6" aria-hidden="true" />
                      )}
                  </Disclosure.Button>
                </div>
              </div>
            </div>

            <Disclosure.Panel className="md:hidden">
              <div className="px-2 pt-2 pb-3 space-y-1 sm:px-3">
                {navigation.map((item) => (
                  <Disclosure.Button
                    key={item.name}
                    as="a"
                    href={item.href}
                    className={classNames(
                      item.current
                        ? 'bg-gray-900 text-white'
                        : 'text-gray-300 hover:bg-gray-700 hover:text-white',
                      'block px-3 py-2 rounded-md text-base font-medium'
                    )}
                    aria-current={item.current ? 'page' : undefined}
                  >
                    {item.name}
                  </Disclosure.Button>
                ))}
              </div>
              <div className="pt-4 pb-3 border-t border-gray-700">
                <div className="flex items-center px-5">
                  <div className="flex-shrink-0">
                    <img
                      className="h-10 w-10 rounded-full"
                      src={
                        photoProfile !== '' ? photoProfile : userMock.imageUrl
                      }
                      alt=""
                    />
                  </div>
                  <div className="ml-3">
                    <div className="text-base font-medium leading-none text-white">
                      {userMock.user}
                    </div>
                    <div className="text-sm font-medium leading-none text-gray-400">
                      {email !== '' ? email : userMock.email}
                    </div>
                  </div>
                  <Tooltip
                    content={email}
                  >
                    <div
                      className="flex bg-gray-800 p-1 rounded-full text-team-green hover:text-white focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-offset-gray-800 focus:ring-white sm:invisible"
                    >
                      <span className="sr-only">View notifications</span>
                      <p>Bienvenido {alias}</p>
                      <UserIcon className="h-6 w-6  rounded-md" aria-hidden="true" />
                    </div>
                  </Tooltip>
                </div>
                <div className="mt-3 px-2 space-y-1">
                  {userNavigation.map((item) => (
                    <Disclosure.Button
                      key={item.name}
                      as="a"
                      href={item.href}
                      className="block px-3 py-2 rounded-md text-base font-medium text-gray-400 hover:text-white hover:bg-gray-700"
                    >
                      {item.name}
                    </Disclosure.Button>
                  ))}
                </div>
              </div>
            </Disclosure.Panel>
          </>
        )}
      </Disclosure>
    </>
  )
}
