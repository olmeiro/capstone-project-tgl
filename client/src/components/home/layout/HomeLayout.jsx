import React from 'react'
import PropTypes from 'prop-types'

import { FooterPage } from './Footer'
import { NavBar } from './NavBar'

export const HomeLayout = ({ children }) => {
  return (
    <div className="bg-purple-100">
      <NavBar />
      {children}
      <FooterPage />
    </div>
  )
}

HomeLayout.propTypes = {
  children: PropTypes.node
}
