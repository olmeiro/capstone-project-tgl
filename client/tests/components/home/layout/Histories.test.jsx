import { configureStore } from '@reduxjs/toolkit'
import { Provider } from 'react-redux'
import { MemoryRouter } from 'react-router-dom'
import { render, screen } from '@testing-library/react'
import { Histories } from '../../../../src/components/home/layout/Histories'
import { useHomeStore } from '../../../../src/hooks/useHomeStore'
import { homeSlice } from '../../../../src/store'

const store = configureStore({
  reducer: {
    home: homeSlice.reducer
  }
  // preloadedState: {
  //   home: {

  //   }
  // }
})

jest.mock('../../../../src/hooks/useHomeStore')

describe('test component <Histories />', () => {
  test('should render component correctly', () => {
    useHomeStore.mockReturnValue({
      getPostsToHomeHook: jest.fn(),
      likeAPost: jest.fn(),
      getInfoFromTheUserLoggedIn: jest.fn(),
      getFavoritesHook: jest.fn()
    })
    const { container } = render(
      <Provider store={store}>
        <MemoryRouter>
          <Histories />
        </MemoryRouter>
      </Provider>
    )
    screen.debug()
    expect(container).toMatchSnapshot()
  })
})
