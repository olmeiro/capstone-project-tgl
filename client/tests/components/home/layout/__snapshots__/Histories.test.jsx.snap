// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test component <Histories /> should render component correctly 1`] = `
<div>
  <div
    class="flex flex-col basis-1 md:basis-1/3 scroll-smooth"
  />
</div>
`;
